lua/neogit/status.lua

```diff
@@ -692,33 +692,28 @@ end
 ---@param first_line number
 ---@param last_line number
 ---@param partial boolean
----@return SelectedHunk[],string[]
+---@return SelectedHunk[]
 function M.get_item_hunks(item, first_line, last_line, partial)
   if item.folded or item.hunks == nil then
-    return {}, {}
+    return {}
   end
 
   local hunks = {}
-  local lines = {}
 
   for _, h in ipairs(item.hunks) do
-    -- Transform to be relative to the current item/file
-    local first_line = first_line - item.first
-    local last_line = last_line - item.first
-
-    if h.diff_from <= last_line and h.diff_to >= first_line then
```

Comment:
Right, I think I know what the problem is.

This calculation treated the cursor position from the items linenr in the statusbuffer as the line number in the diff.

This is correct if each diff is laid out after one another like with `git diff`, but when folding the numbers no longer match up.

Reply:
Using `.first` which from what I've understood mean `rendered line number of this item in the status buffer` is correct here